packages/tabs/src/tab-nav.vue

```diff
@@ -0,0 +1,376 @@
+<script lang='ts'>
+import { h, defineComponent, ref, inject, computed, onUpdated, onMounted, onBeforeUnmount } from 'vue'
+import { addResizeListener, removeResizeListener } from '@element-plus/utils/resize-event'
+import { firstUpperCase } from '@element-plus/utils/util'
+import TabBar from './tab-bar.vue'
+import { NOOP } from '@vue/shared'
+
+export default defineComponent({
+  name: 'ElTabNav',
+  components: {
+    TabBar,
+  },
+  props: {
+    panes: {
+      type: Array,
+      default: () => ([]),
```

Comment:
As I mentioned above, specify the type could help typing system to identify error easier